app/frontend/terrier/tabs.ts

```diff
@@ -51,6 +91,26 @@ export class TabContainerPart extends TerrierPart<TabContainerState> {
             this.state.side = m.data.side
             this.dirty()
         })
+
+        this.makePlugin(SortablePlugin, {
+            zoneClass: 'tt-tab-list',
+            targetClass: 'tab',
+            onSorted: (_, evt) => {
+                this.renumberTabs(evt.toChildren)
+            }
+        })
+    }
+
+    renumberTabs(tabElementsMaybe?: HTMLElement[]) {
+        const tabElements = tabElementsMaybe ??
+            Array.from(this.element?.querySelectorAll('.tt-tab-list') ?? [])
+        const permutation = tabElements.map((tabElement, index) => {
+            const tab = this.tabs[tabElement.dataset.key!]
+            const oldPosition = tab.position
+            tab.position = index
+            return oldPosition
+        })
+        this.emitMessage(TabContainerPart.tabReorderedKey, { permutation })
```

Comment:
Rather than emitting the a permutation, I think it'd be better to just emit an array of the tab keys in the new order.

Something like:

```ts
    renumberTabs(tabElementsMaybe?: HTMLElement[]) {
        if (!tabElementsMaybe) return
        const newTabOrder = tabElements.map(el => el.dataset.key)
        this.emitMessage(this.tabReorderedKey, { newTabOrder })
    }
```

With this approach we don't need to separately keep track of the `position` on each tab, and the user of the tab container doesn't need to know how to use a permutation, they just get a new ordering in the domain they're already working in.